server/src/main/java/org/elasticsearch/rest/ChunkedRestResponseBody.java

```diff
@@ -43,6 +47,30 @@ public interface ChunkedRestResponseBody {
      */
     boolean isDone();
 
+    /**
+     * @return {@code true} if this is the last chunked body in the response, or {@code false} if the REST layer should request further
+     * chunked bodies by calling {@link #getContinuation}.
+     */
+    boolean isEndOfResponse();
+
+    /**
+     * <p>Asynchronously retrieves the next part of the body. Called if {@link #isEndOfResponse} returns {@code false}.</p>
+     *
+     * <p>Note that this is called on a transport thread, so implementations must take care to dispatch any nontrivial work elsewhere.</p>
+
+     * <p>Note that the {@link Task} corresponding to any invocation of {@link Client#execute} completes as soon as the client action
+     * returns its response, so it no longer exists when this method is called and cannot be used to receive cancellation notifications.
+     * Instead, if the HTTP channel is closed while sending a response (including while waiting for a continuation) then the REST layer
+     * will invoke {@link RestResponse#close}. Implementations will typically explicitly create a {@link CancellableTask} to represent the
```

Comment:
> (including while waiting for a continuation)

Hmm on reflection this bit seems implausible. Maybe it only happens the next time we supply a continuation. Needs a test at least.

Reply:
Yep this bit was wrong, adjusted to match reality.